stupidtable.js

```diff
@@ -1,94 +1,121 @@
+
+    // No datatype? Nothing to do.
+    if (datatype === null) {
+      return;
+    }
+
+    // Account for colspans
+    $this_th.parents("tr").find("th").slice(0, $(this).index()).each(function() {
+      var cols = $(this).attr("colspan") || 1;
+      th_index += parseInt(cols,10);
+    });
+
+    var sort_dir;
+    if(arguments.length == 1){
+        sort_dir = force_direction;
+    }
+    else{
+        sort_dir = force_direction || $this_th.data("sort-default") || dir.ASC;
+        if ($this_th.data("sort-dir"))
+           sort_dir = $this_th.data("sort-dir") === dir.ASC ? dir.DESC : dir.ASC;
+    }
+
+
+    $table.trigger("beforetablesort", {column: th_index, direction: sort_dir});
+
+    // More reliable method of forcing a redraw
+    $table.css("display");
+
+    // Run sorting asynchronously on a timout to force browser redraw after
+    // `beforetablesort` callback. Also avoids locking up the browser too much.
+    setTimeout(function() {
+      // Gather the elements for this column
+      var column = [];
+      var sortFns = $table.data('sortFns');
+      var sortMethod = sortFns[datatype];
+      var trs = $table.children("tbody").children("tr");
+
+      // Extract the data for the column that needs to be sorted and pair it up
+      // with the TR itself into a tuple. This way sorting the values will
+      // incidentally sort the trs.
+      trs.each(function(index,tr) {
+        var $e = $(tr).children().eq(th_index);
+        var sort_val = $e.data("sort-value");
+
+        // Store and read from the .data cache for display text only sorts
+        // instead of looking through the DOM every time
+        if(typeof(sort_val) === "undefined"){
+          var txt = $e.text();
+          $e.data('sort-value', txt);
+          sort_val = txt;
+        }
+        column.push([sort_val, tr]);
+      });
+
+      // Sort by the data-order-by value
+      column.sort(function(a, b) { return sortMethod(a[0], b[0]); });
+      if (sort_dir != dir.ASC)
+        column.reverse();
+
+      // Replace the content of tbody with the sorted rows. Strangely
+      // enough, .append accomplishes this for us.
+      trs = $.map(column, function(kv) { return kv[1]; });
+      $table.children("tbody").append(trs);
+
+      // Reset siblings
+      $table.find("th").data("sort-dir", null).removeClass("sorting-desc sorting-asc");
+      $this_th.data("sort-dir", sort_dir).addClass("sorting-"+sort_dir);
+
+      $table.trigger("aftertablesort", {column: th_index, direction: sort_dir});
+      $table.css("display");
+    }, 10);
+
+    return $this_th;
+  };
+
+  // Call on a sortable td to update its value in the sort. This should be the
+  // only mechanism used to update a cell's sort value. If your display value is
+  // also your sort value, omit the second argument. new_display_val is HTML
+  // friendly. Assumes stupidtable has already been called for the table.
+  $.fn.updateSortVal = function(new_display_val, new_sort_val){
```

Comment:
I'm considering separating this out into two separate functions because if the display value is large, say a `<select>` element, I wouldn't want the user to incur the overhead of removing/updating large bodies of text every time they just want the sort value updated. As is, if you want to update the sort value, you must also update the display value. I forsee people having to query the DOM to get the display value and pass that in as an argument, which would be an unfortunate inefficiency. So yeah...I'm probably gonna fix that now.